agent/engine/docker_task_engine.go

```diff
@@ -616,43 +627,105 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 	containerMap, ok := engine.state.ContainerMapByArn(task.Arn)
 	if !ok {
 		return DockerContainerMetadata{
-			Error: CannotStartContainerError{fmt.Errorf("Container belongs to unrecognized task %s", task.Arn)},
+			Error: CannotStartContainerError{errors.Errorf("Container belongs to unrecognized task %s", task.Arn)},
 		}
 	}
 
 	dockerContainer, ok := containerMap[container.Name]
 	if !ok {
 		return DockerContainerMetadata{
-			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
+			Error: CannotStartContainerError{errors.Errorf("Container not recorded as created")},
 		}
 	}
 	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
 }
 
 func (engine *DockerTaskEngine) provisionContainerResources(task *api.Task, container *api.Container) DockerContainerMetadata {
-	seelog.Infof("Task [%v]: Setting up container resources for container [%v]", task.String(), container.String())
+	seelog.Infof("Task [%s]: Setting up container resources for container [%s]", task.String(), container.String())
+	cniConfig, err := engine.buildCNIConfigFromTaskContainer(task, container)
+	if err != nil {
+		return DockerContainerMetadata{
+			Error: ContainerNetworkingError{errors.Wrap(err, "container resource provisioning: unable to build cni configuration")},
+		}
+	}
+	// Invoke the libcni to config the network namespace for the container
+	err = engine.cniClient.SetupNS(cniConfig)
+	if err != nil {
+		log.Error("Set up pause container namespace failed, err: %v, task: %s", err, task.String())
```

Comment:
Please use seelog here. The error message can be `"Set up pause container namespace failed for task [%s]: %v"`